File: Costumed.py
import sys
sys.stdin = open("Costumed.in", "r")
people: list[tuple] = []
suspects: list[tuple] = []

def match(a: tuple[str, str, str], b: tuple[str, int, str]) -> bool:
    persIsUnderClass: bool = a[0] in ["Skeleton", "Zombie", "Witch"]
    susIsUnderClass: bool = b[1] < 11
    persIsBoy: bool = a[1] == "Pumpkin"
    susIsBoy: bool = b[2] == "Boy"
    persIsTaylor: bool = a[2] in ["Kit-Kats", "Jolly Ranchers", "Hershey Kisses"]
    susIsTaylor: bool = b[0] == "Taylor"
    return (persIsUnderClass == susIsUnderClass) and (persIsBoy == susIsBoy) and (persIsTaylor == susIsTaylor)

np: int = int(input())
for i in range(np):
    lp: list[str] = input().split()
    costume: str = lp[0]
    holding: str = lp[1]
    snack: str = " ".join(lp[2:])
    people.append((costume, holding, snack))

ns: int = int(input())
for i in range(ns):
    ls: list[str] = input().split()
    name: str = ls[0]
    school: str = " ".join(ls[1: -2])
    grade: int = int(ls[-2])
    gender: str = ls[-1]
    suspects.append((school, grade, gender))

for i in range(len(people)):
    try: 
        person: tuple[str, str, str] = people[i]
        matches: bool = False
        for suspect in suspects:
            if match(person, suspect):
                matches = True
                break
        if not matches:
            people.pop(i)
            i -= 1
    except:
        break

print(len(people))

File: Costumed.java
import java.util.*;
import java.io.*;

class tuple<X, Y, Z> { 
    public X x; 
    public Y y; 
    public Z z;
    public tuple(X x, Y y, Z z) { 
      this.x = x; 
      this.y = y; 
      this.z = z;
    } 
  } 

public class Costumed {
        public static boolean match(tuple<String, String, String> a, tuple<String, Integer, String> b) {
            boolean persIsUnderClass = Arrays.asList("Skeleton", "Zombie", "Witch").contains(a.x);
            boolean susIsUnderClass = b.y < 11;
            boolean persIsBoy = a.y.equals("Pumpkin");
            boolean susIsBoy = b.z.equals("Boy");
            boolean persIsTaylor = Arrays.asList("Kit-Kats", "Jolly Ranchers", "Hershey Kisses").contains(a.z);
            boolean susIsTaylor = b.x.equals("Taylor");
            return (persIsUnderClass == susIsUnderClass) && (persIsBoy == susIsBoy) && (persIsTaylor == susIsTaylor);
        }

        public static void main(String[] args) throws FileNotFoundException{
        Scanner sc = new Scanner(new File("Costumed.in"));
        ArrayList<tuple<String, String, String>> people = new ArrayList<>();
        ArrayList<tuple<String, Integer, String>> suspects = new ArrayList<>();

        int np = sc.nextInt();
        sc.nextLine();
        for (int i = 0; i < np; i++) {
            String[] lp = sc.nextLine().split(" ");
            String costume = lp[0];
            String holding = lp[1];
            String snack = String.join(" ", Arrays.copyOfRange(lp, 2, lp.length));
            people.add(new tuple<String, String, String>(costume, holding, snack));
        }

        int ns = sc.nextInt();
        sc.nextLine();
        for (int i = 0; i < ns; i++) {
            String[] ls = sc.nextLine().split(" ");
            String name = ls[0];
            String school = String.join(" ", Arrays.copyOfRange(ls, 1, ls.length - 2));
            int grade = Integer.parseInt(ls[ls.length - 2]);
            String gender = ls[ls.length - 1];
            suspects.add(new tuple<String, Integer, String>(school, grade, gender));
        }

        for (int i = 0; i < people.size(); i++) {
            try {
                tuple<String, String, String> person = people.get(i);
                boolean matches = false;
                for (tuple<String, Integer, String> suspect : suspects) {
                    if (match(person, suspect)) {
                        matches = true;
                        break;
                    }
                }
                if (!matches) {
                    people.remove(i);
                    i--;
                }
            } catch (Exception E) {
                break;
            }
        }
        System.out.println(people.size());
    }
}